Searchbar across all bib files instead each having its own

bartsch-dev opened the pull request:
Implemented #825.
The SearchBar is now in the top Toolbar instead of each Tab having its own (always at the right).
On Tab change the Tab gets updated.
The Autocomplete is bound to the current Tab (maybe extend this one in another PR?).
The old global Search (opening all findings in a new Window) is gone, do we really need it?

![jabref_globalsearch](https://cloud.githubusercontent.com/assets/15333371/16581743/8b88f5fc-42ab-11e6-8d65-e2b0802d968c.png)

- [X] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [X] Screenshots added (for bigger UI changes)

boceckts commented on src/main/java/net/sf/jabref/gui/JabRefFrame.java:
```diff
@@ -622,22 +624,26 @@ public void windowClosing(WindowEvent e) {
+                globalSearchBar.setVisible(false);
+                globalSearchBar.setAutoCompleter(null);
                 return;
             }
```
This only disables the global search bar when the last database will be closed. But the search bar will be shown when one is starting jabref with no open database.

bartsch-dev replied:
good catch, I'll fix it quickly

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -0,0 +1,267 @@
+        setLayout(new WrapLayout(FlowLayout.RIGHT));
+        toolBar.setFloatable(false);
+        toolBar.add(currentResults);
+        toolBar.add(searchIcon);
```
I would remove this since the textfield already displays a search icon.

boceckts replied:
I guess this only affects mac os x as it automatically adds a icon when you write
``
if (OS.OS_X) {
    searchField.putClientProperty("JTextField.variant", "search");
}
``

tschechlovdev commented on src/main/java/net/sf/jabref/logic/search/SearchQueryHighlightObservable.java:
```diff
@@ -20,19 +20,17 @@
-            listeners.add(l);
+        if (!listeners.contains(newListener)) {
+            listeners.add(newListener);
+            newListener.highlightPattern(pattern);
```
what happens if listener.contains(newListener) ? should't there be an else block with return?

bartsch-dev replied:
There is, the logic is entirely the same as before, I just rearranged it (as I think it's  more understandable) and added the Builder pattern.

boceckts commented on src/main/java/net/sf/jabref/logic/search/SearchQueryHighlightObservable.java:
```diff
@@ -20,19 +20,17 @@
-     * @param l SearchQueryHighlightListener to be added
+     * @param newListener SearchQueryHighlightListener to be added
      */
-    public void addSearchListener(SearchQueryHighlightListener l) {
```
Please rename the object.

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -0,0 +1,267 @@
+        searchField.addKeyListener(new KeyAdapter() {
+            @Override
+            public void keyReleased(KeyEvent e) {
+                if (e.getExtendedKeyCode() == KeyEvent.VK_ESCAPE) {
```
Please use the `KeyBinding.CLEAR_SEARCH` key.

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -0,0 +1,271 @@
+        }
+
+        SearchQuery searchQuery = getSearchQuery();
+        LOGGER.debug("Searching " + searchQuery + " in " + currentBasePanel.getTabTitle());
```
`LOGGER.debug()` should be used to log error messages, here it seems more like a status update.

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -0,0 +1,268 @@
+
+        setLayout(new WrapLayout(FlowLayout.RIGHT));
+        toolBar.setFloatable(false);
+        toolBar.add(currentResults);
```
Is it really necessary to display the found results in front of the search field? It takes up a lot of space in the toolbar and when having a small resoultion or small screen it might not even be visible as shown below.
![screen shot 2016-07-05 at 14 28 16](https://cloud.githubusercontent.com/assets/15340757/16584748/cec164a0-42bd-11e6-9680-959aa41d1e66.png)

So why not update the status text at the very bottom of the main frame?

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -117,12 +114,14 @@ public GlobalSearchBar() {
-                    endSearch();
-                }
+            public void actionPerformed(ActionEvent e) {
+                System.out.println(endSearch);
```
Remove this line please.

boceckts commented:
Good work 👍

teertinker commented:
I am really happy that you are working on this. Is it correct, that the "global search" window will be abandoned? I have no idea whether it is causing any conflict with the current implementation of the search mechanism, but I am one of those who is using it rather extensively. 

There are two scenarios, where a global search is useful:

1. Often I create a separate bib file for a project (e.g. a journal article), where I put all related papers. Some of them are of interest just for this project and are not supposed to be part of my database core (e.g. newspaper articles). Hence, I often use search queries that include results of the core and project related file

2. I have a bib file for papers which I use in a certain context, but that haven't been read thoroughly yet. Some of them will eventually be moved to the 'core database'. Also in this case I use a global search to find results within both bib-files.

Both scenarios could be managed by creating groups, and if global search gets abandoned, I will certainly do. However, I like the tab based style of several bib files very much, because I don't have the feeling that my project related articles get lost in my huge database with thousands of entries

koppor commented:
:+1: for keeping the global search window. Please readd it. @bartsch-dev There are many JabRef users having more than 10 bib files. They do not want to click through all 10 files for each search.

matthiasgeiger commented:
Another aspect: What happens if the horiztonal space is not enough to display all icons and the search bar? Will the searchbar be displayed beneath the icon bar?

oscargus commented:
Based on another issue the minimum width of the window is already around 1000 pixels (depending on look-an-feel, OS, etc), so it is crucial that it will fold.

But it indeed looks promising!

bartsch-dev commented:
if the window is too small for the toolbar it will now wrap between the normal buttons and the searchbar.

The search result Dialog will now remember its location; the global search has a shortcut (ctrl+shift+f, #573); enter in the result dialog will select the entry in the BasePanel (#617); the Reuslt window is cloasable with ctrl+w if fdocused (https://github.com/JabRef/jabref/issues/573#issuecomment-232284156); and the result dialog won't stay at top anymore if the MainFrame is focused and they are on top of each other.

boceckts commented on src/main/java/net/sf/jabref/gui/keyboard/KeyBinding.java:
```diff
@@ -67,15 +64,14 @@
     SAVE_DATABASE("Save database", Localization.lang("Save database"), "ctrl S"),
-    SAVE_DATABASE_AS(
-            "Save database as ...", Localization.lang("Save database as..."), "ctrl shift S"),
+    SAVE_DATABASE_AS("Save database as ...", Localization.lang("Save database as..."), "ctrl shift S"),
```
Please do not change the format of lines you didn't modify.

boceckts commented:
Please resolve conflicts.

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -90,7 +93,7 @@ public GlobalSearchBar() {
 
         openCurrentResultsInDialog.setToolTipText(Localization.lang("Show search results in a window"));
         openCurrentResultsInDialog.addActionListener(ae -> {
```
Please rename the variable "ae", same goes for the other listeners named "l"

boceckts commented on src/main/java/net/sf/jabref/gui/search/SearchResultsDialog.java:
```diff
@@ -296,6 +316,36 @@ private void addEntry(BibEntry entry, BasePanel panel) {
     }
 
+    private void selectEntryInBasePanel(BibEntry entry){
+        BasePanel p = entryHome.get(entry);
```
Please rename variable.

boceckts commented:
> enter in the result dialog will select the entry in the BasePanel

+ I have two databases open and search for a keyword
+ I open the global search and it finds my result entries from both databases
+ I press enter on an entry and it focuses the correct tab but <b>it always selects the first entry in this tab</b>

Did I missunderstand the sentence or is this a bug?

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -16,201 +16,203 @@
+    private final JTextFieldWithUnfocusedText searchField = new JTextFieldWithUnfocusedText(Localization.lang("Search") + "...");
+    private final JButton clearSearchButton = new JButton(IconTheme.JabRefIcon.CLOSE.getSmallIcon());
+    private final JButton openCurrentResultsInDialog = new JButton(IconTheme.JabRefIcon.OPEN_IN_NEW_WINDOW.getSmallIcon());
+    private final JButton globalSearch = new JButton(Localization.lang("Search globally"));
```
Can you find an icon for this as well? Toolbars shouldn't include buttons with large strings.

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -16,201 +16,203 @@
-        searchModeButton.setToolTipText(searchMode.getToolTipText());
+        searchModeButton.setText(searchDisplayMode.getDisplayName());
+        searchModeButton.setToolTipText(searchDisplayMode.getToolTipText());
     }
```
Can you use icons for this as well?

bartsch-dev replied:
I can't find good Icons for this.

bartsch-dev commented:
Should be fixed now!

boceckts commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -16,201 +16,202 @@
         toolBar.add(new HelpAction(HelpFiles.SEARCH));
-
+        toolBar.addSeparator();
+        toolBar.add(currentResults);
```
This will cause the toolbar to jump around when typing and clearing a search query. To fit the searchbar in the first row I need to use full screen mode and than there is no space to the right at all. So once I start typing the result text will be displayed and the searchbar jumps to the second row, when I clear the search it jumps back to the 1st row since the text is invisible. In my opinion the text with the number of found results should be somewhere else.

bartsch-dev replied:
I've set the preffered size of the lable to 100px which contains the standard `Found X entries`-Message.

koppor commented on CHANGELOG.md:
```diff
@@ -16,11 +16,14 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 - [#1026](https://github.com/JabRef/jabref/issues/1026) JabRef does no longer delete user comments outside of BibTeX entries and strings
 - [#1249](https://github.com/JabRef/jabref/issues/1249) Date layout formatter added
+- Implemented [#825](https://github.com/JabRef/jabref/issues/825): Search Bar across all bib files instead each having its own
+- Implemented [#573](https://github.com/JabRef/jabref/issues/573): Add key shortcut for global search
```
Which shortcut?

bartsch-dev replied:
I set it to `strg+shift+f` which opens the global search resutl dialog.

koppor commented:
Is it possible to work on the global search in this PR, too? 

- [ ] The shortcut ctrl+shift+f should focus the search field if it is empty (instead of doing a globally searching the empty string)
- [ ] ESC in the global search should focus the search field. Currently, the focus just stays where it was
- [ ] The global search popup should show the bib file name

bartsch-dev commented:
implemented all 3 things.

boceckts commented on src/main/java/net/sf/jabref/model/entry/BibEntry.java:
```diff
@@ -199,6 +199,7 @@ public void setType(EntryType type) {
      * Returns the contents of the given field, or null if it is not set.
+     * @deprecated use {@link #getFieldOptional} instead
      */
     @Deprecated //Use getFieldOptional instead
```
This comment is now unnecessary, please remove it.

simonharrer commented:
What happens if the window is too small? Will the search bar be moved to the new line? A lot of people use JabRef only with 1024x768 resolution.

bartsch-dev commented:
Yes, if the width is too small the entire searchbar will move into a new line.

bartsch-dev commented:
What I've done so far:

- SearchBar
  - instead of each bib database having its own, the searchbar is now in the Toolbar
  - if the window width is too small the whole searchbar will move into a new line
  - added shortcut: global search `ctrl+shift+f`; if the search field is empty it focus the searchbar

- SearchResultWindow
  - the window will now remember its location
  - each entry now shows which bib file it belongs to
  - when the user closes the window (without selecting an entry) the searchbar is focused
  - when the user double clicks on an entry it will correctly be selected in the bib file
  - `enter` will select the entry in the bib file  (#617)
  - added shortcut: closing the window `ctrl+w` (https://github.com/JabRef/jabref/issues/573#issuecomment-232284156)
  - the result window won't stay on top anymore the main frame is focused


What do you think?

matthiasgeiger commented:
Sounds good :wink:

I'll test it later and give you some feedback

matthiasgeiger commented:
Good job! Almost all your described functionality is working fine. :+1: 

The exception is "fixed: when searching the first match will be selected if the current selection is no match" - this still is not working for me in "filter" mode.

Some ideas for further improvments:
- the database column in the result window should be longer - the current width is determined by the word "database" and only the first 9 characters of the filename is shown.
- in windows re-accessing the result window after switching the focus to another application is hard as it is not added to the windows task bar (was no issue before as it was modal 'above' the JabRef main window)
- I'm not sure whether it might be helpful to use a (titled?) border to group the search related textfields, buttons and labels (see https://docs.oracle.com/javase/tutorial/uiswing/components/border.html) - or maybe align the search stuff on the right side?

bartsch-dev commented:
> The exception is "fixed: when searching the first match will be selected if the current selection is no match" - this still is not working for me in "filter" mode.

When no entry was selected at all the search wouldn't select a match, it now does. Did you mean that?

> the database column in the result window should be longer - the current width is determined by the word "database" and only the first 9 characters of the filename is shown.

The database column is now as wide as the author column.

> in windows re-accessing the result window after switching the focus to another application is hard as it is not added to the windows task bar (was no issue before as it was modal 'above' the JabRef main window)

The SearchResultFrame will now show in the Taskbar (you can focus it with the global search shortkey too).

> ...  or maybe align the search stuff on the right side? 

I wanted to do that too, but if I do that the toolbar is not wrapable anymore.

matthiasgeiger commented:
Your PR is currently broken:

```
Exception in thread "AWT-EventQueue-0" java.lang.ArrayIndexOutOfBoundsException: 4
        at net.sf.jabref.gui.search.SearchResultFrame.setWidths(SearchResultFrame.java:305)
        at net.sf.jabref.gui.search.SearchResultFrame.init(SearchResultFrame.java:155)
        at net.sf.jabref.gui.search.SearchResultFrame.<init>(SearchResultFrame.java:135)
        at net.sf.jabref.gui.search.GlobalSearchWorker.<init>(GlobalSearchWorker.java:35)
        at net.sf.jabref.gui.search.GlobalSearchBar.performGlobalSearch(GlobalSearchBar.java:206)
        at net.sf.jabref.gui.search.GlobalSearchBar.lambda$new$1(GlobalSearchBar.java:117)
        at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
...
```

bartsch-dev commented:
Sadly I cannot reproduce your problem!

bartsch-dev commented:
I will be gone the next 2 weeks. I will be continuing working then.

bartsch-dev commented:
What needs to be done for this PR to be closed?

koppor commented on CHANGELOG.md:
```diff
@@ -46,6 +53,16 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
 ### Fixed
 - Fixed [#1632](https://github.com/JabRef/jabref/issues/1632): User comments (@Comment) with or without brackets are now kept
 - Preview panel height is now saved immediately, thus is shown correctly if the panel height is changed, closed and opened again
+### Fixed           
```
Why this addition?

bartsch-dev replied:
Must have overlooked it while rebasing.
fixed

koppor commented on CHANGELOG.md:
```diff
@@ -46,6 +53,16 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
+- Entries in the SearchResultDialog are now converted to Unicode
+- suggestions in the autocomplete (search) are now in unicde
+- Fixed [#617](https://github.com/JabRef/jabref/issues/617): Enter in search dialog window opens the selected entry
+- Fixed [#1632](https://github.com/JabRef/jabref/issues/1632) User comments (@Comment) with or without brackets are now kept
```
Remove from your changes, it is already there (line 54)

bartsch-dev replied:
Must have overlooked it while rebasing.
fixed

koppor commented on CHANGELOG.md:
```diff
@@ -46,6 +53,16 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
+- Fixed: When searching the first match will be selected if the current selection is no match
+- Selecting an entry in the search result Window will now select the correct entry in the bib file
+- Entries in the SearchResultDialog are now converted to Unicode
+- suggestions in the autocomplete (search) are now in unicde
```
Start with capital letter

bartsch-dev replied:
fixed

koppor commented on src/main/java/net/sf/jabref/gui/search/SearchResultFrame.java:
```diff
@@ -1,11 +1,27 @@
+/*  Copyright (C) 2003-2016 JabRef contributors.
```
Remove the GPL header

bartsch-dev replied:
fixed

koppor commented on src/main/java/net/sf/jabref/logic/search/rules/RegexBasedSearchRule.java:
```diff
@@ -51,9 +52,13 @@ public boolean applyRule(String query, BibEntry bibEntry) {
-                String fieldContent = RegexBasedSearchRule.REMOVE_LATEX_COMMANDS
-                        .format(bibEntry.getFieldOptional(field).get());
-                String fieldContentNoBrackets = RegexBasedSearchRule.REMOVE_LATEX_COMMANDS.format(fieldContent);
+                Optional<String> fieldOptional = bibEntry.getFieldOptional(field);
```
Merge with the previous line:

    Optional<String> fieldOptional = bibEntry.getFieldOptional(field);
    if (fieldOptional.isPresent()) {
       ... remaining code...

Then, you also don't need the check on line 56.

Siedlerchr replied:
Or you can simply use:
```
bibEntry.getFieldOptional.ifPresent(field->  {
do stuff 
} );
```
Then the else would be not necessary, because it would automatically continue with the next entry

bartsch-dev replied:
The lambda is not possible b/c of the `return` (or is it?).

koppor commented on src/main/java/net/sf/jabref/gui/entryeditor/EntryEditorTab.java:
```diff
@@ -1,3 +1,18 @@
+/*  Copyright (C) 2003-2016 JabRef contributors.
```
Remove GPL header

bartsch-dev replied:
fixed

koppor commented on src/main/java/net/sf/jabref/gui/maintable/MainTableSelectionListener.java:
```diff
@@ -1,3 +1,18 @@
+/*  Copyright (C) 2003-2016 JabRef contributors.
```
Remove GPL header

bartsch-dev replied:
fixed

koppor commented on src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java:
```diff
@@ -0,0 +1,349 @@
+/*  Copyright (C) 2003-2016 JabRef contributors.
```
Remove GPL header :innocent:

bartsch-dev replied:
fixed

koppor commented:
Please fix my minor comments. @matthiasgeiger Any additional comments? Otherwise, I would merge this in. (After double check CHANGELOG.md and headers)

Siedlerchr commented on src/main/resources/l10n/JabRef_da.properties:
```diff
@@ -1760,3 +1760,4 @@ One_file_found=
 The_import_finished_with_warnings\:=
 There_was_one_file_that_could_not_be_imported.=
 There_were_%0_files_which_could_not_be_imported.=
+Global_search=
```
I don't know where it exactly is used, but we already have a resource for `Search_globally=Search_globally`

bartsch-dev replied:
fixed

Siedlerchr commented on src/main/java/net/sf/jabref/gui/search/SearchResultFrame.java:
```diff
@@ -103,14 +126,22 @@
-        diag = new JDialog(frame, title, false);
+        searchResultFrame = new JFrame();
+        searchResultFrame.setTitle(title);
+        searchResultFrame.setIconImage(new ImageIcon(IconTheme.getIconUrl("jabrefIcon48")).getImage());
```
Simpler: Use IconTheme.getImageIcon( )

bartsch-dev replied:
fixed

Siedlerchr commented on src/main/java/net/sf/jabref/gui/search/SearchResultFrame.java:
```diff
@@ -460,36 +538,40 @@ public Object getColumnValue(BibEntry entry, int column) {
-                    if (entry.hasField(FieldName.URL)) {
-                        urlLabel.setToolTipText(entry.getFieldOptional(FieldName.URL).get());
+                case URL_COL: {
+                    Optional<String> urlField = entry.getFieldOptional(FieldName.URL);
```
Simpler: e`ntry.getFieldOptional(FieldName.URL).ifPresent(urlLabel::setToopTipText)`
The latter one is a nice way of writing method references in lambdas, see here for an overview:
https://blog.idrsolutions.com/2015/02/java-8-method-references-explained-5-minutes/

bartsch-dev replied:
They are there to define the scope for the `width` variable which are both in `DATABASE_COL` and `default`.

Siedlerchr commented on src/main/java/net/sf/jabref/gui/search/SearchWorker.java:
```diff
@@ -60,44 +60,52 @@ protected void done() {
-        for (BibEntry entry : matchedEntries) {
-            entry.setSearchHit(true);
-        }
+        basePanel.getDatabase().getEntries().forEach(entry -> entry.setSearchHit(false));
```
Some devs don't like this for each version, so please use the old one.

bartsch-dev replied:
What exactly is the reason that they do not like the new version?

Siedlerchr replied:
This applies only to the for each, because it's harder to debug. We had a discussion about it once. @koppor

bartsch-dev replied:
fixed

Siedlerchr commented on src/main/java/net/sf/jabref/gui/search/SearchWorker.java:
```diff
@@ -60,44 +60,52 @@ protected void done() {
+        List<BibEntry> selectedEntries = basePanel.getSelectedEntries();
+        boolean isHitSelected = !selectedEntries.isEmpty();
+        for (BibEntry entry : selectedEntries) {
+            if (!entry.isSearchHit()) {
```
You could simpler use streams: `isHitSelected = selectedEntries.stream().anyMatch(  condtion ) `

bartsch-dev replied:
done

koppor commented:
Could you please make `Search globally` make a toggle button and change the functionally of the button before to behave acoordingly?

If the search globally is on, the global search is active. Otherwise, the search string is local for each database.

matthiasgeiger commented:
You can also use the icon `earth` (Codepoint F1E7) and `earth-off` (F1E8;) for this.

oscargus commented:
As I was just writing something similar as @matthiasgeiger , here are the images for reference:

![earth](https://cloud.githubusercontent.com/assets/8114497/17968273/2fc2d1a4-6acc-11e6-8a75-6717c32fee4b.png)
![earth-off](https://cloud.githubusercontent.com/assets/8114497/17968272/2fc1e708-6acc-11e6-96ec-ba5cebc7a14a.png)

bartsch-dev commented:
added a global mode to the search

- if the global mode is off
  - each tab has its own searchterm
  - the search result dialog only shows local findings

- if the global mode is on
  - in each tab the searchterm is the same (taking the one when it was activated)
  - the search result dialog shows findings in all bib files

mlep commented:
Does the help file need an update?

bartsch-dev commented:
The global search needs to be added to the help file.
The `show results in window` functionality needs to be described too.

Braunch commented:
What is the current status on this? Is there anything @bartsch-dev should do here?

boceckts commented:
Looking at the localization files and the changelog it seems like you have some merge problems.
Fixing the localization files should fix travis CI. Also try to fix codacy/pr issues.

oscargus commented on src/main/java/net/sf/jabref/gui/IconTheme.java:
```diff
@@ -157,6 +157,8 @@
         FACEBOOK("\uf20c"), /* css: facebook */
         BLOG("\uf46b"), /* css: rss */
+        GLOBAL_SEARCH_ON("\uF1E7"),
+        GLOBAL_SEARCH_OFF("\uF1E8"),
```
Can you please add a comment here about the css names? `earth` and `earth-off`

Will make it easier if they start moving the icons around in the font...

bartsch-dev replied:
done

koppor commented:
I've got an issue with the focus. Discussed personally with @bartsch-dev. Would merge after that is fixed.

bartsch-dev commented:
I fixed the focus problem.
I also addressed a traversal problem after you abort the searcht (hit <kbd>ESC</kbd> in the searchbar).
Now the User can jump from the searchbar to the Main table with <kbd>ctrl</kbd> + <kbd>Enter</kbd>

koppor commented on CHANGELOG.md:
```diff
@@ -12,6 +12,13 @@ We refer to [GitHub issues](https://github.com/JabRef/jabref/issues) by using `#
+- The search result Window will now show which entry belongs to which bib file
+- The search result Window will now remember its location
+- The search result Window won't stay on top anymore if the main Window is focused and will be present in the taskbar
+- The User can jump from the searchbar to the maintabel with `ctrl + enter`
```
fix typos `user`, `maintable`

koppor replied:
Keyboard shortcuts should be rendered consistenly without space before and after `+`

bartsch-dev replied:
fixed

koppor commented on src/main/resources/l10n/JabRef_da.properties:
```diff
@@ -1781,3 +1780,4 @@ should_have_the_first_letter_capitalized=
 Remember_password?=
 
 shared=
+
```
Would you mind not adding an empty line to prevent merge conflicts?

bartsch-dev replied:
fixed

koppor commented:
Searchbar disappers at certain occaisions. Please try to increase/decrease the width of JabRef and you'll see that at some point, it disappears:

![grabbed_20160908-224223](https://cloud.githubusercontent.com/assets/1366654/18366072/86d91e9a-7615-11e6-84f9-81e71f2ebb89.png)

![grabbed_20160908-224233](https://cloud.githubusercontent.com/assets/1366654/18366080/8cd6214e-7615-11e6-8696-6453ac4f92a2.png)

It also happens during the search if "no results found" is appended. Then, the searchbar disapears after having nothing found:

![grabbed_20160908-224028](https://cloud.githubusercontent.com/assets/1366654/18366014/443e09b0-7615-11e6-9978-41331842a1b3.png)

![grabbed_20160908-224043](https://cloud.githubusercontent.com/assets/1366654/18366024/4b7496c2-7615-11e6-884d-fae815de5d89.png)

bartsch-dev commented:
fixed

koppor commented:
Tested with bce1bf5 - not fixed:

![jabref-global-search](https://cloud.githubusercontent.com/assets/1366654/18416353/69048144-7812-11e6-99e7-7f61db3c51ba.gif)
